src/app/extensions/store-locator/pages/store-locator/store-locator-page.component.ts

```diff
@@ -0,0 +1,118 @@
+import { ChangeDetectionStrategy, Component, OnInit, ViewChild } from '@angular/core';
+import { FormGroup } from '@angular/forms';
+import { FormlyFieldConfig } from '@ngx-formly/core';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+
+import { AppFacade } from 'ish-core/facades/app.facade';
+import { Country } from 'ish-core/models/country/country.model';
+
+import { StoreLocatorFacade } from '../../facades/store-locator.facade';
+import { StoreHelper } from '../../models/store/store.helper';
+import { Store as StoreModel } from '../../models/store/store.model';
+import { StoresMapComponent } from '../../shared/stores-map/stores-map.component';
+
+@Component({
+  selector: 'ish-store-locator-page',
+  templateUrl: './store-locator-page.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class StoreLocatorPageComponent implements OnInit {
+  @ViewChild('map') map: StoresMapComponent;
+
+  loaded$: Observable<boolean>;
+  stores$: Observable<StoreModel[]>;
+  highlightedStore$: Observable<StoreModel>;
+
+  form = new FormGroup({});
+  model = { countryCode: '', postalCode: '', city: '' };
+  fields: FormlyFieldConfig[];
+
+  constructor(private storeLocatorFacade: StoreLocatorFacade, private appFacade: AppFacade) {}
+
+  ngOnInit() {
+    this.fields = this.getFields();
+    this.loaded$ = this.storeLocatorFacade.getLoaded$();
+    this.stores$ = this.storeLocatorFacade.getStores$();
+    this.highlightedStore$ = this.storeLocatorFacade.getHighlighted$();
+  }
+  private getFields() {
+    return [
+      {
+        key: 'countryCode',
+        type: 'ish-select-field',
+        wrappers: ['form-field-horizontal'],
+        templateOptions: {
+          required: false,
+          label: 'store_locator.form.country.label',
+          placeholder: 'store_locator.form.country.default',
+          options: this.appFacade
+            .countries$()
+            ?.pipe(
```

Comment:
`appFacade.countries$()` always returns an observable, and even if that doesn't contain any values, it should never be falsy. Instead, you should probably move the optional chaining operator to `countries.map`, because the first emission could very well be falsy.

Reply:
I don't even know, why I put it there in the first place. I think it was because I ran into some kind of error but I couldn't replicate any of it with the current version. Think this can safely be removed now